src/userinput.c

```diff
@@ -135,6 +135,7 @@ static int pinentry_read(int from, char **retstr)
 	    || strncmp(buf, "D ", 2) == 0) {
 		if (retstr) {
 			*retstr = strchr(buf, ' ');
+			**retstr = *retstr ? strtok_r(*retstr, "\n", &saveptr) : NULL;
```

Comment:
For a single call to `strtok_r` I believe you don't need `saveptr`, isn't `NULL` enough as the 3rd argument?

Also this is suspect:
```
**retstr = *retstr [...]
```
Perhaps is should be:
```
*retstr = *retstr [...]
```